### PR TITLE
`run-tests` as Python script

### PR DESCRIPTION
**Pull request by rmitsch**
## Description

Replaces `run-tests.sh` with `run-tests.py` for maintenance reasons. The new script is functionally mostly equivalent, except for improved logging.
Note: this is branched off, and should be merged after, https://github.com/explosion/projects/pull/121.

### Types of change

Enhancement.

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [ ] I ran the tests, and all new and existing tests passed.
- [ ] I ran the `update` scripts in the `.github` folder, and all the configs and docs are up-to-date.

### CODE REVIEW

File: run-tests.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        elif cmds_status[-1]["returncode"] == 5:
+            _log(logger, verbosity, "vv", "    No tests found")
+
+        _log(logger, verbosity, "v", "  - Restoring environment")
+        with tempfile.NamedTemporaryFile("w") as file:
+            file.writelines(
+                _run(
+                    [python, "-m", "pip", "freeze", "--exclude", "torch", "cupy-cuda111"], cmds_status
+                ).stdout.decode("utf-8").split("\n")
+            )
```
Comment (polm) [inline run-tests.py:127]: `writelines` does not add line separators, so the input here is run together and the resulting file is one line. 

```suggestion
            file.write(
                _run(
                    [python, "-m", "pip", "freeze", "--exclude", "torch", "cupy-cuda111"], cmds_status
                ).stdout.decode("utf-8")
            )
```

**Comment by adrianeboyd** [comment]
Can you rebase this on the current `v3`?

**Comment by adrianeboyd** [comment]
Can you revert all the changes beyond `run-tests.py`?

**Comment by rmitsch** [comment]
> Can you revert all the changes beyond `run-tests.py`?

Reverted formatting changes, the remaining changes are critical to this PR. (Note that this is not working yet, cf. our internal discussion about problems with executing this from an Python interpreter).

File: experimental/ner_wikiner_speedster/requirements.txt [hunk 2]
```diff
@@ -1,2 +1,3 @@
 speedster>=0.0.1
 spacy-transformers>=1.1.0,<1.2.0
+huggingface-hub>=0.2.0,<1.0.0
```
Comment (adrianeboyd) [inline experimental/ner_wikiner_speedster/requirements.txt:3]: This doesn't make sense?

**Comment by svlandeg** [comment]
Closing and adding to our internal backlog.

**Pull request closed without merging**